### PR TITLE
bears/hypertext: Add HTTPoliceLintBear

### PR DESCRIPTION
**Pull request by mr-karan**
This bear uses `httpolice` to lint
har files.

Closes https://github.com/coala-analyzer/coala-bears/issues/596

### CODE REVIEW

File: .ci/deps.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ go get -u github.com/golang/lint/golint
 go get -u golang.org/x/tools/cmd/goimports
 go get -u sourcegraph.com/sqs/goreturns
 go get -u golang.org/x/tools/cmd/gotype
-go get -u github.com/kisielk/errcheck 
+go get -u github.com/kisielk/errcheck
```
Comment (Adrianzatreanu) [inline .ci/deps.sh]: unrelated to commit

File: .ci/deps.sh [hunk 2]
```diff
@@ -84,7 +84,7 @@ julia -e "Pkg.add(\"Lint\")"
 sudo luarocks install luacheck --deps-mode=none
 
 # Infer commands
-if [ ! -e ~/infer-linux64-v0.7.0/infer/bin ]; then
```
Comment (Adrianzatreanu) [inline .ci/deps.sh]: unrelated to commit

File: bears/hypertext/HTTPoliceLintBear.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+from coalib.settings.Setting import typed_list
+
+
+@linter(executable='httpolice',
+        output_format='regex',
+        output_regex=r'-*\s*(?P<origin>(request|response).*)\n'
+        r'(?P<severity>E|C|D)\s(?P<message>.*)')
+class HTTPoliceLintBear:
+    """
+    HTTPolice is a lint for HTTP requests and responses. It checks them for
```
Comment (Adrianzatreanu) [inline bears/hypertext/HTTPoliceLintBear.py]: i think its called ``linter`` :D lint i think is the verb

File: bears/hypertext/HTTPoliceLintBear.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    conformance to standards and best practices.
+
+    For more information on the analysis check
+    <https://github.com/vfaronov/httpolice>.
+    """
+    LANGUAGES = {"HAR"}
```
Comment (Adrianzatreanu) [inline bears/hypertext/HTTPoliceLintBear.py]: from what ive googled, HAR is a format not a language, you sure its appropiate?

File: requirements.txt [hunk 5]
```diff
@@ -13,6 +13,8 @@ cpplint==1.*
 isort==4.*
 html-linter==0.3.*
 # Do *not* use 0.3 which is not backwards compatible to common clang versions
+lxml==3.4.2
```
Comment (Adrianzatreanu) [inline requirements.txt]: what is this and if its for the bear, why is it not in the REQUIREMENTS tuple?

File: bears/hypertext/HTTPoliceLintBear.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.Setting import typed_list
+
+
+@linter(executable='httpolice',
+        output_format='regex',
+        output_regex=r'-*\s*(?P<origin>(request|response).*)\n'
+        r'(?P<severity>E|C|D)\s(?P<message>.*)')
```
Comment (AbdealiJK) [inline bears/hypertext/HTTPoliceLintBear.py]: Indentation - we use:

```
def a(
    a="aa"
      "bb")
```

File: bears/hypertext/HTTPoliceLintBear.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax'}
+    severity_map = {'E': RESULT_SEVERITY.MAJOR,
+                    'C': RESULT_SEVERITY.NORMAL,
+                    'D': RESULT_SEVERITY.INFO}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         silent: typed_list(str)=[]):
+        silent = tuple('-s='+part for part in silent)
```
Comment (AbdealiJK) [inline bears/hypertext/HTTPoliceLintBear.py]: Doc comment for `silent`

**Comment by Adrianzatreanu** [comment]
ping @mr-karan

**Comment by sils** [comment]
hey @mr-karan do you intend to clean up your stuff? :(

**Comment by yash-nisar** [comment]
Carried forward in https://github.com/coala/coala-bears/commit/40dd5d4c14b32140314df19f5b52d14cb03e7c84. @Makman2 we need to close this :wink:
